airbyte-integrations/connectors/source-oracle-v2/src/main/kotlin/io/airbyte/integrations/source/oracle/OracleSourceConfigurationJsonObject.kt

```diff
@@ -0,0 +1,270 @@
+package io.airbyte.integrations.source.oracle
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter
+import com.fasterxml.jackson.annotation.JsonAnySetter
+import com.fasterxml.jackson.annotation.JsonGetter
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.annotation.JsonPropertyDescription
+import com.fasterxml.jackson.annotation.JsonPropertyOrder
+import com.fasterxml.jackson.annotation.JsonSetter
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import com.fasterxml.jackson.annotation.JsonValue
+import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaArrayWithUniqueItems
+import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaDefault
+import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaDescription
+import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaInject
+import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaTitle
+import io.airbyte.cdk.command.CONNECTOR_CONFIG_PREFIX
+import io.airbyte.cdk.command.ConnectorConfigurationJsonObjectBase
+import io.airbyte.cdk.ssh.MicronautPropertiesFriendlySshTunnelMethodConfigurationJsonObject
+import io.airbyte.cdk.ssh.SshTunnelMethodConfiguration
+import io.airbyte.commons.exceptions.ConfigErrorException
+import io.micronaut.context.annotation.ConfigurationBuilder
+import io.micronaut.context.annotation.ConfigurationProperties
+import jakarta.inject.Singleton
+
+/**
+ * The object which is mapped to the Oracle source configuration JSON.
+ *
+ * Use [OracleSourceConfiguration] instead wherever possible.
+ * This object also allows injecting values through Micronaut properties,
+ * this is made possible by the classes named `MicronautPropertiesFriendly.*`.
+ */
+@JsonSchemaTitle("Oracle Source Spec")
+@JsonPropertyOrder(
+    value =
+    [
+        "host",
+        "port",
+        "connection_data",
+        "username",
+        "password",
+        "schemas",
+        "jdbc_url_params",
+        "encryption",
+    ],
+)
+@Singleton
+@ConfigurationProperties(CONNECTOR_CONFIG_PREFIX)
+class OracleSourceConfigurationJsonObject : ConnectorConfigurationJsonObjectBase() {
+
+    @JsonProperty("host", required = true)
```

Comment:
Might not work with our Static `Jsons.<ser/deser>` calls but I'm exploring (still don't have fully working one) [jackson's support for Kotlin](https://mvnrepository.com/artifact/com.fasterxml.jackson.module/jackson-module-kotlin) by injecting objectMapper to the caller.

Reply:
[Working model](https://github.com/airbytehq/airbyte/commit/70c5c81d47fa8f71fb641d97e9b427f115f0cf94#diff-3a6fe5c2dceac061ed2789191ec903bb39bf34a80b04cf47babf8099dd1ad74aR7)  if you want to try data classes for immutability.

Reply:
Thanks! I definitely prefer immutable objects.

Here that's tempered by the fact that the serializable config object is rarely what we want the business logic to interact with directly. In the case of source-oracle-v2, we actually prefer `OracleSourceConfiguration`. There are definitely config components, like the ssh configuration, which are simple enough to be directly used, so your point definitely applies.

There's a constraint to bear in mind: we want to be able to inject config values via micronaut properties. This might make the serializable object a bit ugly anyway.

I'm hardly a jackson expert and I'm pretty keen to simplify these object definitions so let's revisit this together when the time is right!

Reply:
W.r.t injecting a mapper: can we add the `KotlinModule` to the object mapper used by `Jsons`? If possible then there's no need to inject custom object mappers.

Reply:
yeah i don't see any harm adding it to `MoreMappers.initMappers` (famous last words)